### ISSUE TITLE
Worst Case Scenario Submission Issues.

### ISSUE THREAD

**Issue by MKoppJones** [hmrc/vat-api#216]
Hi,

We are currently thinking about our worst case scenarios.

We have built out submission around a 3 step process:
1 - Query for the period key
2 - Perform the submission
3 - Query for the period key and check that it has been fulfilled

At step 2, we get a receipt ID. In a case where HMRC fails to respond in a reasonable time, or there is a timeout, it is possible that our submission was successful.

Is there a way of retrieving the receipt after the fact?

**Comment by MarkAKelly** [comment]
Hi @MKoppJones,

TL;DR: As it stands there is not a way to retrieve the receipt ID.  

When you say HMRC fails to respond, you will mean in this instance that the request to the VAT API through the API platform has timed out.  If this is the case, then it will most likely be the case that the submission hasn't been made, meaning there won't be a receipt ID.  

If however, there are any timeouts or failures to submit resulting from any of HMRC's backend systems (those responsible for processing the submission and that sit behind our API), we would handle this and return a suitable error and message (usually an ISE).  In this circumstance, it may well be the case that once the request has left our service the submission is then processed, however the Receipt-ID won't be bundled into our response header as we cant acknowledge that as a successful request.

It's worth noting that the Receipt-ID is more of a ticket of acknowledgement of the Vat Submission.  It is a reference number passed to us by a HMRC service that logs the time and information in the request, rather than a receipt of the actual processing of the data.  The submission is proved by the Form Bunde Ref which is returned to us as part of the response body of a successful submission.  As far as I am aware I don't think there is a way to retrieve either of these after a submission.

Hope this makes things a little clearer, feel free to come back to us on anything if it doesn't!

Mark

**Comment by MKoppJones** [comment]
This makes sense regarding the Receipt ID.

Will there ever be a case where the submission is successful but timesout on a response?

In that case, is there a way to get the Form Bundle Ref afterwards?

**Comment by MarkAKelly** [comment]
There is always that possibility.  I wouldn't like to say that it would never happen, but if it did then it would be a small chance.  

Again unfortunately, in this instance there is way to access the Form Bundle ref after the fact via an API call.

Mark

**Comment by MKoppJones** [comment]
We are aware that the chances are slim and might not happen, just trying to cover ourselves, thank you for the help.

**Comment by TCardinal** [comment]
It would be very useful to be able to retrieve the response data for a submission - receipt id, form bundle, even maybe the correlation id at some point after the submission.
